processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -1026,10 +1051,21 @@ public boolean matches()
       }
     }
 
-    // Currently unused, except by unit tests, since filters always support bitmap indexes currently.
-    // This will change when non-String dimensions are added.
     @Override
-    public ValueMatcher makeValueMatcher(String dimension, final Predicate predicate)
+    public ValueMatcher makeValueMatcher(String dimension, final DruidPredicateFactory predicateFactory)
+    {
+      ValueType type = getTypeForDimension(dimension);
+      switch (type) {
+        case LONG:
+          return makeLongValueMatcher(dimension, predicateFactory.makeLongPredicate());
+        case STRING:
+          return makeStringValueMatcher(dimension, predicateFactory.makeStringPredicate());
+        default:
+          throw new UOE("Cannot make ValueMatcher for type[%s]", type);
```

Comment:
Hmm, this should probably act like the column is a string column full of nulls. The rationale is that the classic Druid behavior is that when you filter on a column that doesn't support filtering, it acts as if the column is full of nulls. So that'd be consistent with behavior in 0.9.1.1 and prior.

This could be done by calling `predicateFactory.makeStringPredicate()` and checking if it applies to `null` or not, then using that to make a BooleanValueMatcher.